python/metatensor-learn/metatensor/learn/nn/layer_norm.py

```diff
@@ -0,0 +1,321 @@
+            normalized_shape, torch.Size
+        ):
+            normalized_shape = [normalized_shape] * len(in_keys)
+        elif not (isinstance(normalized_shape, List)):
+            raise TypeError(
+                "`normalized_shape` must be integer or List of integers,"
+                f" but not {type(normalized_shape)}."
+            )
+        elif len(in_keys) != len(normalized_shape):
+            raise ValueError(
+                "`normalized_shape` must have same length as `in_keys`, but"
+                f" len(normalized_shape) != len(in_keys)"
+                f" [{len(normalized_shape)} != {len(in_keys)}]"
+            )
+
+        for n_shape in normalized_shape:
+            if not (
+                isinstance(n_shape, int)
+                or isinstance(n_shape, List)
+                or isinstance(n_shape, torch.Size)
+            ):
+                raise TypeError(
+                    "`normalized_shape` must be integer or List of"
+                    f" integers, but not {type(n_shape)}."
+                )
+        # Check `eps`
+        if isinstance(eps, float):
+            eps = [eps] * len(in_keys)
+        elif not (isinstance(eps, List)):
+            raise TypeError(
+                "`eps` must be float or List of float, but not"
+                f" {type(normalized_shape)}."
+            )
+        elif len(in_keys) != len(eps):
+            raise ValueError(
+                "`eps` must have same length as `in_keys`, but"
+                f" len(eps) != len(in_keys) [{len(eps)} !="
+                f" {len(in_keys)}]"
+            )
+        # Check `elementwise_affine`
+        if isinstance(elementwise_affine, bool):
+            elementwise_affine = [elementwise_affine] * len(in_keys)
+        elif not (isinstance(elementwise_affine, List)):
+            raise TypeError(
+                "`elementwise_affine` must be float or List of float,"
+                f" but not {type(normalized_shape)}."
+            )
+        elif len(in_keys) != len(elementwise_affine):
+            raise ValueError(
+                "`elementwise_affine` must have same length as `in_keys`,"
+                f" but len(elementwise_affine) != len(in_keys)"
+                f" [{len(elementwise_affine)} != {len(in_keys)}]"
+            )
+        # Check `bias`
+        if isinstance(bias, bool):
+            bias = [bias] * len(in_keys)
+        elif not (isinstance(bias, List)):
+            raise TypeError(
+                "`bias` must be float or List of float, but not"
+                f" {type(normalized_shape)}."
+            )
+        elif len(in_keys) != len(bias):
+            raise ValueError(
+                "`bias` must have same length as `in_keys`, but"
+                f" len(bias) != len(in_keys) [{len(bias)} !="
+                f" {len(in_keys)}]"
+            )
+
+        # Build module list
+        modules = []
+        for i in range(len(in_keys)):
+            module = torch.nn.LayerNorm(
```

Comment:
Shouldn't we pass the `elementwise_affine`, `bias`, `device` and `dtype` to the `torch.nn.LayerNorm` constructor?

Reply:
Good spot, thanks!